backend/service/topology/querybuilder_test.go

```diff
@@ -0,0 +1,337 @@
+		},
+		{
+			id:        "Page 0 with limit set",
+			filter:    &topologyv1.SearchRequest_Filter{},
+			sort:      &topologyv1.SearchRequest_Sort{},
+			pageToken: "0",
+			limit:     5,
+			expect:    "SELECT id, data, metadata FROM topology_cache ORDER BY ID ASC LIMIT 5 OFFSET 0",
+		},
+		{
+			id:        "Change PageToken and Limits",
+			filter:    &topologyv1.SearchRequest_Filter{},
+			sort:      &topologyv1.SearchRequest_Sort{},
+			pageToken: "10",
+			limit:     5,
+			expect:    "SELECT id, data, metadata FROM topology_cache ORDER BY ID ASC LIMIT 5 OFFSET 50",
+		},
+		{
+			id: "All Options",
+			filter: &topologyv1.SearchRequest_Filter{
+				Search: &topologyv1.SearchRequest_Filter_Search{
+					Field: "metadata.search.field",
+					Text:  "cat",
+				},
+				TypeUrl: "type.googleapis.com/clutch.aws.ec2.v1.AutoscalingGroup",
+				Metadata: map[string]string{
+					"label": "value",
+				},
+			},
+			sort: &topologyv1.SearchRequest_Sort{
+				Field:     "metadata.meow.iam.a.cat",
+				Direction: topologyv1.SearchRequest_Sort_ASCENDING,
+			},
+			pageToken: "10",
+			limit:     5,
+			expect:    "SELECT id, data, metadata FROM topology_cache WHERE quote_literal(metadata->'search'->'field') LIKE $1 AND resolver_type_url = $2 AND metadata @> $3::jsonb ORDER BY $4 ASC LIMIT 5 OFFSET 50",
+		},
+	}
+	for _, test := range testCases {
+		output, _, err := paginatedQueryBuilder(test.filter, test.sort, test.pageToken, test.limit)
+		assert.NoError(t, err)
+
+		sql, _, err := output.ToSql()
+		assert.NoError(t, err)
+		assert.Equal(t, test.expect, sql)
+	}
+}
+
+func TestFilterQueryBuilder(t *testing.T) {
+	testCases := []struct {
+		id     string
+		input  *topologyv1.SearchRequest_Filter
+		expect string
+	}{
+		{
+			id:     "No Input",
+			input:  &topologyv1.SearchRequest_Filter{},
+			expect: "SELECT * FROM topology_cache",
+		},
+		{
+			id: "Search by column",
+			input: &topologyv1.SearchRequest_Filter{
+				Search: &topologyv1.SearchRequest_Filter_Search{
+					Field: "column.id",
+					Text:  "cat",
+				},
+			},
+			expect: "SELECT * FROM topology_cache WHERE quote_literal(id) LIKE $1",
+		},
+		{
+			id: "Search by Metadata",
+			input: &topologyv1.SearchRequest_Filter{
+				Search: &topologyv1.SearchRequest_Filter_Search{
+					Field: "metadata.label",
+					Text:  "cat",
+				},
+			},
+			expect: "SELECT * FROM topology_cache WHERE quote_literal(metadata->>'label') LIKE $1",
+		},
+		{
+			id: "Search all options",
+			input: &topologyv1.SearchRequest_Filter{
+				Search: &topologyv1.SearchRequest_Filter_Search{
+					Field: "metadata.label",
+					Text:  "cat",
+				},
+				TypeUrl: "type.googleapis.com/clutch.aws.ec2.v1.AutoscalingGroup",
+				Metadata: map[string]string{
+					"label":  "value",
+					"label2": "value2",
+				},
+			},
+			expect: "SELECT * FROM topology_cache WHERE quote_literal(metadata->>'label') LIKE $1 AND resolver_type_url = $2 AND metadata @> $3::jsonb",
```

Comment:
the where clause still appears to be vulnerable to injection, it doesn't look like Expr prevents it